### PR TITLE
Login/Signup make backend request

### PR DESCRIPTION
**Pull request by Wennxn**
#16 
This issue is bigger than I thought, might have to break it into 2 phases.

DONE:
- changed Login/Sign Up forms to class components
- make a request to login with username password
- make a request to signup a new user

TODO
Backend:
/signup: check if email/phonenumber/username already existed in the database
Frontend:
- Login: hash plain text password before sending request 
- Login + SignUp: generate error msg when user inputs are missing

### CODE REVIEW

File: screens/LoginScreen.js [hunk 1]
```diff
@@ -1,57 +1,122 @@
...
+  }
+
+  onLoginButtonPress() {
+    const { username, password } = this.state;
+    console.log("[DEBUG] LOGIN Button pressed.");
+    console.log("[DEBUG] username is " + username + ", password is " + password);
+
+    this.setState({ loading: true });
+
+    axios.post(SERVER_ROOT + '/login', {
```
Comment (Asiapenguin) [inline screens/LoginScreen.js]: Extract requests to a service

Comment (Asiapenguin) [inline screens/LoginScreen.js]: Extract requests to a service

File: App_test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// import React, { useState } from 'react';
```
Comment (Asiapenguin) [inline App_test.js]: what is this file?

Comment (Wennxn) [reply App_test.js]: that's the React hooks that I commented out, should've removed it

File: screens/LoginScreen.js (hunk 1 above)
Comment (Wennxn) [reply screens/LoginScreen.js]: what requests?

**Comment by Asiapenguin** [comment]
Also, no longer need to hash password before making request to signup/login, we will change to oauth2 before releasing. This is just a stub for development purposes

**Pull request merged**